src/client/gl-base-technique.cc

```diff
@@ -79,6 +80,23 @@ GlBaseTechnique::Commit()
     update_rendering_queue_->Push(std::move(command));
   }
 
+  if (!pending_.gl_textures.empty()) {
+    std::list<std::pair<uint32_t, std::weak_ptr<GlTexture>>> gl_textures;
+    pending_.gl_textures.swap(gl_textures);
+    auto command = CreateCommand([gl_textures = std::move(gl_textures),
+                                     rendering = rendering_](bool cancel) {
+      if (cancel) {
+        return;
+      }
+
+      for (auto& texture : gl_textures) {
+        rendering->gl_textures.push_back(texture);
+      }
```

Comment:
Please remove the texture in rendering->gl_textures when
- the weak_ptr doesn't own a resource.
- the location of the texture is the same as one of the newly pushing textures. (which means texture is overwritten)